### PR TITLE
added a few more endpoints

### PR DESCRIPTION
**Pull request by casakawa-jama**
added in some more endpoints:
- **get server response** i wanted an endpoint to do a priliminary check to see if the server creds are gewd
- **get item** gets a singular item
- **get item types** return all the item types
- **get item type** gets a singular item type
- **get children items** returns all the children of a specified ID
- **get upstream relationships** gets the all the upstream relationships (not items) of a specified ID
- **get downstream relationships** gets the all the downstream relationships (not items) of a specified ID

### CODE REVIEW

**Comment by casakawa-jama** [comment]
not sure if `get_server_resposne()` belongs in the client..

File: py_jama_rest_client/client.py [hunk 1]
```diff
@@ -54,6 +54,12 @@ def __init__(self, host_domain, credentials=('username', 'password'), api_versio
         self.__credentials = credentials
         self.__core = Core(host_domain, credentials, api_version=api_version, oauth=oauth)
 
+    def get_available_endpoints(self):
+        """Returns a list of all the available endpoints."""
```
Comment (nmchale-jama) [inline py_jama_rest_client/client.py]: Please use google style comment blocks for functions. 
EX: https://gist.github.com/dsaiztc/9b4c15c6dfef08a957c1

Comment (nmchale-jama) [reply py_jama_rest_client/client.py]: doing this will allow the auto doc tools to correctly produce API documentation.

**Comment by casakawa-jama** [comment]
added in even more.

- get pick lists
- get pick list
- get pick list options

**Pull request merged**